woocommerce/woocommerce-gateway-stripe
# Capture off-session payments immediately so we can fetch charge information

Fixes #3274

## Changes proposed in this Pull Request:

In v8.5.0 of Stripe we bumped the Stripe API version to `2024-06-20`. One effect of that is the default payment intent `capture_method` was changed to async (see Stripe [changelog here](https://docs.stripe.com/upgrades#2024-04-10)). This means that when we create payment intents without a `capture_method` the response won't include a charge that has a balance translation. Instead Stripe will charge the payment asynchronisly and attach the charge later. 

This change causes the subscription renewal flow to no longer receive the balance transaction information immediately and therefore not assign Stripe fees via [`process_response()`](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/fix/issue-3274-capture-subscription-renewal-immediately/includes/abstracts/abstract-wc-stripe-payment-gateway.php#L544-L547).

This PR fixes that by making sure we specify `capture_method => automatic` so the payment is charged immediately and the response includes a completed charge. 

## Testing instructions

1. Purchase a subscription product using Stripe. 
2. From the **WooCommerce > Subscriptions** list table view the subscription you just purchased.
3. From the actions drop down select **"Process Renewal"**
4. Save the subscription. 
5. View the newly created renewal order. 
   - On `develop` you'll notice that there are no Stripe fees.
   - On this branch fees will be displayed. 

| `develop` | This branch |
|--------|--------|
| <img width="462" alt="Screenshot 2024-07-17 at 7 35 02 PM" src="https://github.com/user-attachments/assets/5640761b-2814-40c4-9f4a-02cb845b87ea"> | <img width="335" alt="Screenshot 2024-07-17 at 7 37 51 PM" src="https://github.com/user-attachments/assets/2be2e79d-e541-4523-97c1-bd46f40c146f"> | 

---


**Post merge**
